### PR TITLE
DAITDS-143 - Pulsar Job support to Terraform

### PR DESCRIPTION
**Pull request by ThiagoYU**

### CODE REVIEW

File: ns1/resource_pulsar_job.go [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+	s := map[string]*schema.Schema{
+		"customer": {
+			Type:     schema.TypeInt,
+			Computed: true,
+		},
+		"name": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"typeid": {
```
Comment (Zach-Johnson) [inline ns1/resource_pulsar_job.go:25]: I think `type_id` would be easier to read

File: ns1/resource_pulsar_job.go [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validateTypeId,
+		},
+		"community": {
+			Type:     schema.TypeBool,
+			Computed: true,
+		},
+		"jobid": {
```
Comment (Zach-Johnson) [inline ns1/resource_pulsar_job.go:34]: same thing, consider making this `job_id`

File: ns1/resource_pulsar_job.go [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+			Type:     schema.TypeString,
+			Computed: true,
+			ForceNew: true,
+		},
+		"appid": {
```
Comment (Zach-Johnson) [inline ns1/resource_pulsar_job.go:39]: consider making this `app_id`

File: ns1/resource_pulsar_job.go [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+
+	return pulsarJobToResourceData(d, &j)
+}
+
+// pulsarJobRead reads the given zone data from ns1
+func pulsarJobRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ns1.Client)
+	j, resp, err := client.PulsarJobs.Get(d.Get("appid").(string), d.Id())
+	if err != nil {
+		if err == ns1.ErrAppMissing {
```
Comment (Zach-Johnson) [inline ns1/resource_pulsar_job.go:374]: consider using `errors.Is(err, ns1.ErrAppMissing)` instead

File: ns1/resource_pulsar_job.go [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+			log.Printf("[DEBUG] NS1 Pulsar Application (%s) not found", d.Get("appid"))
+			d.SetId("")
+			return nil
+		}
+
+		if err == ns1.ErrJobMissing {
```
Comment (Zach-Johnson) [inline ns1/resource_pulsar_job.go:380]: consider using `errors.Is(err, ns1.ErrJobMissing)` instead

**Review by Zach-Johnson — CHANGES_REQUESTED** [review]
looking good, just noted a few small things

File: ns1/resource_pulsar_job.go [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+		},
+		"active": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+		"shared": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  true,
```
Comment (mvkushnir) [inline ns1/resource_pulsar_job.go:51]: I think this should be default false

File: ns1/resource_pulsar_job.go [hunk 7]
```diff
@@ -0,0 +1,445 @@
...
+			Computed: true,
+		},
+		"jobid": {
+			Type:     schema.TypeString,
+			Computed: true,
+			ForceNew: true,
+		},
+		"appid": {
+			Type:     schema.TypeString,
+			Optional: true,
```
Comment (mvkushnir) [inline ns1/resource_pulsar_job.go:41]: this should be required

File: website/docs/r/pulsar_job.html .markdown [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+`config` support several sub-options related to job configuration:
+
+* `host` - (Required) Hostname where the resource is located.
+* `url_path` - (Required) URL path to be appended to the host.
+* `https` - (Optional) Indicates whether or not to use HTTPS in measurements.
+* `http` - (Optional) Indicates whether or not to use HTTP in measurements.
+* `request_timeout_millis` - (Optional) The amount of time to allow a single job to perform N runs.
+* `job_timeout_millis` - (Optional) The amount of time to allow a single job to perform 1 run.
+* `use_xhr` - (Optional) Indicates wheter or not to use XmlHttpRequest (XHR) when taking measurements.
+* `satuc_values` - (Optional) Indicates wheter or not to skip aggregation for this job's measurements.
```
Comment (mvkushnir) [inline website/docs/r/pulsar_job.html .markdown:57]: Typo? should be `static_values`

**Review by mvkushnir — APPROVED** [review]
👍 Looks good! Just one final comment

**Review by Zach-Johnson — APPROVED** [review]
lgtm!

**Pull request merged**